src/presentation/pages/Edit/Profile/index.tsx

```diff
@@ -25,19 +26,27 @@ function MyProfileEdit() {
   });
 
   useEffect(() => {
-    const idCheck = /^[a-z]+[a-z|0-9|.|_]{4,15}$/;
-    setIsEditConditionPassed({
-      ...isEditConditionPassed,
-      id: idCheck.test(inputId),
-    });
+    (async () => {
+      if (!inputId) return;
+      const idCheck = /^[a-z]+[a-z|0-9|.|_]{4,15}$/;
+      const { isSuccess } = await api.userService.getDuplicationCheck(inputId);
+      const passedId = idCheck.test(inputId) && !isSuccess;
 
-    if (!idCheck.test(inputId)) {
-      setErrorMsg('*영문 소문자, 숫자, 특수문자(._) 4~15자 이내');
-    }
+      setIsEditConditionPassed({
+        ...isEditConditionPassed,
+        id: passedId,
+      });
 
-    if (!/^[a-z]/.test(inputId)) {
-      setErrorMsg('*아이디의 첫 글자는 영문 소문자');
-    }
+      if (!idCheck.test(inputId)) {
+        setErrorMsg('*영문 소문자, 숫자, 특수문자(._) 4~15자 이내');
+      } else if (isSuccess) {
+        setErrorMsg('*중복된 아이디입니다.');
+      }
+
+      if (!/^[a-z]/.test(inputId)) {
+        setErrorMsg('*아이디의 첫 글자는 영문 소문자');
+      }
```

Comment:
편의를 위해 아이디 조건들에 아래와 같이 숫자를 붙여서 말할게요!
1) *영문 소문자, 숫자, 특수문자(._) 4~15자 이내
2) *중복된 아이디입니다.
3) *아이디의 첫 글자는 영문 소문자

1과 3를 만족하지 않는다면 2에 걸릴 일이 없고 3이 첫글자에 대한거라 제일 먼저 만날 수 있는 오류니 아래와 같은 순서로 검사 로직을 바꾸는거 어떤가요??

```suggestion
      if (!/^[a-z]/.test(inputId)) {
        setErrorMsg('*아이디의 첫 글자는 영문 소문자');
      } else if (!idCheck.test(inputId)) {
        setErrorMsg('*영문 소문자, 숫자, 특수문자(._) 4~15자 이내');
      } else if (isSuccess) {
        setErrorMsg('*중복된 아이디입니다.');
      }
```

Reply:
넘넘 좋아요 ㅠㅠㅠ 반영하겠습니다 👍